libs/auth/oauth.go

```diff
@@ -143,6 +144,19 @@ func (a *PersistentAuth) Challenge(ctx context.Context) error {
 	return nil
 }
 
+// Best effort to remove url path, query args and fragments from the host
+func (a *PersistentAuth) cleanHost() {
+	parsedHost, err := url.Parse(a.Host)
+	if err != nil {
+		return
+	}
+	parsedHost.RawPath = ""
+	parsedHost.RawQuery = ""
+	parsedHost.Path = ""
+	parsedHost.Fragment = ""
```

Comment:
Instead of setting all fields to empty, you can pick the ones you need and carry them over to a new instance (or even inline, since you only care about the string here): https://github.com/databricks/databricks-sdk-go/blob/41df2f4a5fc6ffb274bb1565ae08f23721bf75d7/config/config.go#L405-L409